From 626f6a667538260b571c329a707c29032c0c7d75 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Sat, 11 Apr 2020 00:04:18 +0300
Subject: [PATCH 1/8] gtk: Remove unnecessary linker flags when targeting mingw

The --export-dynamic flag is a no-op when targeting mingw; when used,
binutils ld prints this warning:
warning: --export-dynamic is not supported for PE+ targets, did you mean --export-all-symbols?

The "--exclude-libs ALL" flag also is unnecessary; as this is an
exe, there's no symbols being exported automatically, so there's no
need to exclude anything.

LLD's COFF/mingw backend doesn't implement these flags, and this
fixes linking ghb.exe with it.
---
 gtk/src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/src/Makefile.am b/gtk/src/Makefile.am
index 68504bae6..07c00d795 100644
--- a/gtk/src/Makefile.am
+++ b/gtk/src/Makefile.am
@@ -115,7 +115,7 @@ ghb_SOURCES = \
 
 if MINGW
 ghb_LDFLAGS = \
-	-mwindows -Wl,--export-dynamic -Wl,--exclude-libs,ALL
+	-mwindows
 else
 ghb_LDFLAGS = \
 	-Wl,--export-dynamic -Wl,--exclude-libs,ALL
-- 
2.17.1

